### PR TITLE
chanserv/bansearch: new module

### PR DESCRIPTION
**Pull request by maxteufel**

### CODE REVIEW

File: modules/chanserv/bansearch.c [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+	}
+
+	struct channel *c = channel_find(channel);
+
+	if (!c)
+	{
+		(void) command_fail(si, fault_nosuch_target, _("\2%s\2 is currently empty."), channel);
+		return;
+	}
+
```
[the remark this answers is not in the document]
Comment (maxteufel) [reply modules/chanserv/bansearch.c]: `MODE #channel` and `MODE #channel +bq` works on `+s` channels as well, so there is nothing to hide.

File: modules/chanserv/bansearch.c [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+		}
+		else if (!cu->modes)
+		{
+			(void) command_success_nodata(si, _("\2%s\2 is moderated and \2%s\2 is not voiced or opped."), channel, tu->nick);
+		}
+	}
+
+	if (c->limit != 0 && c->nummembers >= c->limit)
+	{
+		(void) command_success_nodata(si, _("\2%s\2 is full."), channel);
```
[the remark this answers is not in the document]
Comment (maxteufel) [reply modules/chanserv/bansearch.c]: I'd either add this to the invite-only and password lines as well, or make an implicit assumption that the user isn't `+I` and has neither an `/invite` nor the password.

[the remark this answers is not in the document]
Comment (aaronmdjones) [reply modules/chanserv/bansearch.c]: Fair point.

**Comment by aaronmdjones** [comment]
Merged manually with edits ce33878bdaa0e07b9032125db7f159dc6cd6942a

**Pull request closed without merging**